.github/workflows/compiler-sanitizers.yml

```diff
@@ -0,0 +1,49 @@
+name: compiler-sanitizers
+
+on:
+  push:
+  pull_request:
+    types: [ opened, reopened, synchronize ]
+
+jobs:
+  compiler-sanitizers:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        sanitizer: [ 'address,pointer-compare,pointer-subtract', 'thread' ]
+
+    env:
+      CFLAGS:       "-fno-omit-frame-pointer -fstack-protector-all -fsanitize=${{matrix.sanitizer}},bounds,enum -fsanitize-address-use-after-scope -fsanitize-address-use-after-return=always -fsanitize-recover=address -fsanitize-memory-track-origins=2"
```

Comment:
Looking into the log of the action there is a warning:
```
clang: warning: argument unused during compilation: '-fsanitize-memory-track-origins=2' [-Wunused-command-line-argument]
```
Can it be fixed?

Reply:
> Looking into the log of the action there is a warning:
> 
> ```
> clang: warning: argument unused during compilation: '-fsanitize-memory-track-origins=2' [-Wunused-command-line-argument]
> ```
> 
> Can it be fixed?

Yes, i can address this. It'll just appear to be a little messier in the github actions GUI.

If you prefer the warning be addressed over the GUI looking perfect, it's easy to do.